### ISSUE TITLE
Update e2e tests to take more screenshots

### ISSUE THREAD

**Issue by brandyscarney** [ionic-team/ionic#16715]
# Feature Request

**Ionic version:**

[x] **4.x**

**Describe the Feature Request**
Update our e2e tests (specifically overlay components) to take screenshots of more examples.

**Describe Preferred Solution**
The action sheet splits all of the tests into multiple files:

<img width="299" alt="screen shot 2018-12-12 at 5 53 17 pm" src="https://user-images.githubusercontent.com/6577830/49903990-d4ff7280-fe36-11e8-9387-f9b2856dccd8.png">

However, alert loops through each button id to take the screenshots:

```
test('alert: basic', async () => {
  const page = await newE2EPage({
    url: '/src/components/alert/test/basic?ionic:_testing=true'
  });

  const alerts = [
    ['#basic'],
    ['#longMessage', 'long message'],
    ['#multipleButtons', 'multiple buttons'],
    ['#noMessage', 'no message'],
    ['#confirm', 'confirm'],
    ['#prompt', 'prompt'],
    ['#radio', 'radio'],
    ['#checkbox', 'checkbox']
  ];

  for (const [buttonSelector, message] of alerts) {
    await page.click(buttonSelector);

    const alert = await page.find('ion-alert');
    expect(alert).not.toBe(null);
    await alert.waitForVisible();
    await page.waitFor(250);

    const compare = await page.compareScreenshot(message);
    expect(compare).toMatchScreenshot();
    await alert.callMethod('dismiss');
  }

});
```

Meanwhile other components, such as loading aren't taking screenshots of anything but the basic one. I think we should be consistent here, and update the e2e tests for all of the overlay components to the same style:

- [x] Action Sheet
- [x] Alert
- [x] Loading
- [x] Menu
- [x] Modal
- [x] Picker
- [x] Popover
- [x] Toast

**Comment by brandyscarney** [comment]
cc @manucorporat @camwiegert

**Comment by manucorporat** [comment]
100% agree, i don't like how the action-sheet tests are spread across many fails, it makes it hard to add new tests! I like your format of using a loop, i would modify it to actually move the `test()` inside the loop itself, so we get better idea of what is being tested!

```
describe('alert: basic', async () => {
  const page = await newE2EPage({
    url: '/src/components/alert/test/basic?ionic:_testing=true'
  });

  const alerts = [
    ['#basic'],
    ['#longMessage', 'long message'],
    ['#multipleButtons', 'multiple buttons'],
    ['#noMessage', 'no message'],
    ['#confirm', 'confirm'],
    ['#prompt', 'prompt'],
    ['#radio', 'radio'],
    ['#checkbox', 'checkbox']
  ];

  for (const [buttonSelector, message] of alerts) {
    test(`click ${buttonSelector}`, async () => {
      await page.click(buttonSelector);

      const alert = await page.find('ion-alert');
      expect(alert).not.toBe(null);
      await alert.waitForVisible();
      await page.waitFor(250);

      const compare = await page.compareScreenshot(message);
      expect(compare).toMatchScreenshot();
      await alert.callMethod('dismiss');
    }
  }
});
```

**Comment by brandyscarney** [comment]
In looking to fix this I found a few problems:
- You can't loop over async tests in Jest, it all needs to be in one test block to loop or generated from a script, see https://github.com/facebook/jest/issues/1619
- When the expect fails on the screenshot, it stops taking the remaining screenshots - so if you have a mismatch on the first screenshot, take the alert for example, you won't get the remaining 7 screenshots in the test.

^ The last one is actually a much bigger problem for the tests because even when it fails on expects like the following it won't continue: https://github.com/ionic-team/ionic/blob/master/core/src/components/input/test/basic/e2e.ts#L15. This would mean that that screenshot won't ever be taken. So we need to somehow catch all of the errors from expects but continue and then throw them all at the end. 

cc @manucorporat @adamdbradley

**Comment by liamdebeasi** [comment]
Resolved by #17534
